workspaces/libnpmpublish/lib/provenance.js

```diff
@@ -1,68 +1,206 @@
+            GITHUB_RUN_ATTEMPT: env.GITHUB_RUN_ATTEMPT,
+            GITHUB_RUN_ID: env.GITHUB_RUN_ID,
+            GITHUB_SHA: env.GITHUB_SHA,
+            GITHUB_WORKFLOW_REF: env.GITHUB_WORKFLOW_REF,
+            GITHUB_WORKFLOW_SHA: env.GITHUB_WORKFLOW_SHA,
           },
-          entryPoint: workflowPath,
         },
-        parameters: {},
-        environment: {
-          GITHUB_EVENT_NAME: env.GITHUB_EVENT_NAME,
-          GITHUB_REF: env.GITHUB_REF,
-          GITHUB_REPOSITORY: env.GITHUB_REPOSITORY,
-          GITHUB_REPOSITORY_ID: env.GITHUB_REPOSITORY_ID,
-          GITHUB_REPOSITORY_OWNER_ID: env.GITHUB_REPOSITORY_OWNER_ID,
-          GITHUB_RUN_ATTEMPT: env.GITHUB_RUN_ATTEMPT,
-          GITHUB_RUN_ID: env.GITHUB_RUN_ID,
-          GITHUB_SHA: env.GITHUB_SHA,
-          GITHUB_WORKFLOW_REF: env.GITHUB_WORKFLOW_REF,
-          GITHUB_WORKFLOW_SHA: env.GITHUB_WORKFLOW_SHA,
+        metadata: {
+          buildInvocationId: `${env.GITHUB_RUN_ID}-${env.GITHUB_RUN_ATTEMPT}`,
+          completeness: {
+            parameters: false,
+            environment: false,
+            materials: false,
+          },
+          reproducible: false,
         },
+        materials: [
+          {
+            uri: `git+${env.GITHUB_SERVER_URL}/${env.GITHUB_REPOSITORY}@${env.GITHUB_REF}`,
+            digest: {
+              sha1: env.GITHUB_SHA,
+            },
+          },
+        ],
       },
-      metadata: {
-        buildInvocationId: `${env.GITHUB_RUN_ID}-${env.GITHUB_RUN_ATTEMPT}`,
-        completeness: {
-          parameters: false,
-          environment: false,
-          materials: false,
+    }
+  }
+  if (ci.GITLAB) {
+    payload = {
+      _type: INTOTO_STATEMENT_TYPE,
+      subject,
+      predicateType: SLSA_PREDICATE_TYPE,
+      predicate: {
+        buildType: `${GITLAB_BUILD_TYPE_PREFIX}/${GITLAB_BUILD_TYPE_VERSION}`,
+        builder: { id: `${env.CI_PROJECT_URL}/-/runners/${env.CI_RUNNER_ID}` },
+        invocation: {
+          configSource: {
+            uri: `git+${env.CI_PROJECT_URL}`,
+            digest: {
+              sha1: env.CI_COMMIT_SHA,
+            },
+            entryPoint: env.CI_JOB_NAME,
+          },
+          parameters: {
+            CI: env.CI,
+            CI_API_GRAPHQL_URL: env.CI_API_GRAPHQL_URL,
+            CI_API_V4_URL: env.CI_API_V4_URL,
+            CI_BUILD_BEFORE_SHA: env.CI_BUILD_BEFORE_SHA,
+            CI_BUILD_ID: env.CI_BUILD_ID,
+            CI_BUILD_NAME: env.CI_BUILD_NAME,
+            CI_BUILD_REF: env.CI_BUILD_REF,
+            CI_BUILD_REF_NAME: env.CI_BUILD_REF_NAME,
+            CI_BUILD_REF_SLUG: env.CI_BUILD_REF_SLUG,
+            CI_BUILD_STAGE: env.CI_BUILD_STAGE,
+            CI_COMMIT_AUTHOR: env.CI_COMMIT_AUTHOR,
+            CI_COMMIT_BEFORE_SHA: env.CI_COMMIT_BEFORE_SHA,
+            CI_COMMIT_BRANCH: env.CI_COMMIT_BRANCH,
+            CI_COMMIT_DESCRIPTION: env.CI_COMMIT_DESCRIPTION,
+            CI_COMMIT_MESSAGE: env.CI_COMMIT_MESSAGE,
```

Comment:
On PII, looks like some of these include full name and email, do you want to remove these as well?

```
"CI_COMMIT_AUTHOR": "Full Name <email@chainguard.dev>",
"CI_COMMIT_DESCRIPTION": "\nSigned-off-by: Full Name <email@chainguard.dev>",
"CI_COMMIT_MESSAGE": "hack sigstore-js\n\nSigned-off-by: Full Name <email@chainguard.dev>\n",
```

Reply:
These are from the Git commit, so we wouldn't *really* be able to hide these, but done!